data/transactions/logic/assembler.go

```diff
@@ -1421,7 +1420,7 @@ func typeStores(pgm *ProgramKnowledge, args []string) (StackTypes, StackTypes, e
 	for i := range pgm.scratchSpace {
 		// We can't know what slot stacktop is being stored in, but we can at least keep the slots that are the same type as stacktop
 		if pgm.scratchSpace[i] != pgm.stack[top] {
-			pgm.scratchSpace[i] = StackAny
+			pgm.scratchSpace[i].AVMType = avmAny
```

Comment:
Since you have `unionStackType` now, how about putting the `unionStackType(pgm.scratchSpace[i], pgm.stack[top]` in here?

As written, it bails out way too easily - if you store a bounded integer, like a `bool`, you'll end up wiping out all the integers into `any`, since line 1422 will be true, even though you at least know you have ints.

Reply:
Is what is here now what you meant?

Reply:
Is there an easy way for me to jump to the latest version of this code in PR (even if only approximately)? This seems like a github missing feature. (or a github feature, missed by me)

Reply:
I don't know of a way if there is one 🤔